Harja-905 pudotusvalikkojen nappaimistokaytto

marikork opened the pull request:
Muutettu alasvetolistan avaaminen enterin lisäksi välilyönnillä ja sulkeminen tabilla ja esc:llä (jos on avattu hiirellä).
Alasvetolistan avaamisessa fokus siirretään valittuna olevan tiedon riville, tai jos ei ole mitään valittuna, ylimpään riviin.
Alasvetolistassa liikutaan nuolinäppäimillä, eikä nuolinäppäimistä vielä toteudu rivin valinta.
Rivi valitaan enterillä.
Kun fokus on alasvetolistassa, sen saa kiinni esc:llä, shift:llä tai shift+tabilla, jos ei halua tehdä valintaa.  
Poistettu alasvetovalikon ylämarginaali, koska oli turha väli buttonin ja listan valissä.
Lisätty asettelun seuranta, eli valikko aukeaa ylöspäin, jos alapuolella ei ole koko valikolle tilaa,
mutta yläpuolella on.
Muutettu murupolussa urakoitsija-alasvetovalikon asettelua ja kentän leveyttä.
Murupolussa muutettu myös hallintayksikkö- ja urakka-valikot toimimaan näppäimistöllä.
Muutettu checkbox-alasvetovalikkojen toiminta niin, että valinnan pystyy tekemään enterillä (valikko ei sulkeudu, 
jotta pystyy tekemään lisää valintoja).
Muutamaan kohtaan lisätty viivettä, jotta fokus siirtyisi oikein alasvetovalikosta takaisin buttoniin.
Pariin kohtaan lisätty skrollausasetus.
Muutettu ryhmiteltyä dataa käyttävän pudotusvalikon näppäimistökäyttö menemään oikein.
Lisätty tarkistus, että disabloituja pudotusvalikon rivejä ei pysty käyttämään näppäimistöllä.
Haku-komponentti ei ole vielä näppäimistökäytettävä, se pitää korjata eri tiketissä.

marikork committed:
HARJA-905 Muuta asetteluja sekä valikon avaaminen ja riveillä liikkuminen

marikork committed:
HARJA-905 Muuta asetteluja sekä valikon avaaminen ja riveillä liikkuminen

marikork committed:
HARJA-905 Muuta etusivun hallintayksikön ja urakan alasvetovalikoiden toiminnot

marikork committed:
HARJA-905 Muuta etusivun hallintayksikön ja urakan alasvetovalikoiden toiminnot

marikork committed:
HARJA-905 Muuta checkbox-alasvetovalikkojen näppäimistökäyttö

marikork committed:
HARJA-905 Muuta checkbox-alasvetovalikkojen näppäimistökäyttö

marikork committed:
HARJA-905 Korjaa Kokonaishintaisten näkymässä kenttien fokuksen siirrot

marikork committed:
HARJA-905 Korjaa Kokonaishintaisten näkymässä kenttien fokuksen siirrot

marikork committed:
HARJA-905 Muuta fokuksen siirtyminen kahdessa kohdassa ja lisää yksi skrollausmääritys

marikork committed:
HARJA-905 Muuta fokuksen siirtyminen kahdessa kohdassa ja lisää yksi skrollausmääritys

marikork committed:
HARJA-905 Korjaa valaistusurakan uuden sanktion luomisessa pudotusvalikon asettelu

marikork committed:
HARJA-905 Korjaa valaistusurakan uuden sanktion luomisessa pudotusvalikon asettelu

marikork committed:
HARJA-905 Muuta fokuksen siirto Kulujen kohdistuksessa ja Kustannusten seurannassa

marikork committed:
HARJA-905 Muuta fokuksen siirto Kulujen kohdistuksessa ja Kustannusten seurannassa

samnes committed:
Lisää dom apurit

samnes committed:
Lisää saavutettavuus muutokset modaaliin

samnes committed:
Muokkaa sivupalkki saavutettavaksi

samnes committed:
Lisää gridiin mahdollisuus navigoida tabilla

samnes committed:
Ota saavutettava sivupalkki käyttöön ja korjaa viesti-tiemerkintaan-modal saavutettavaksi

solita-staskila committed:
Yhdenmukaista raporttinappeja sanktiot näkymään

solita-staskila committed:
Sievennä snadisti aikataulu suodatin header tyylejä

solita-staskila committed:
Ryhmitä suodattimia nätimmin, poista turha let, aikataulu

solita-staskila committed:
Poista turhat requiret, yhtenäistä ja ryhmitä raporttipainikkeet

solita-staskila committed:
Poista aria label download ikonista

marikork committed:
HARJA-905 Lisää vaihtoehtojen ryhmittely pudotusvalikon painikkeen tietoihin

marikork committed:
HARJA-905 Lisää vaihtoehtojen ryhmittely pudotusvalikon painikkeen tietoihin

marikork committed:
HARJA-905 Lisää skrollaus kululomakkeen Tehtäväryhmä-pudotusvalikkoon

marikork committed:
HARJA-905 Lisää skrollaus kululomakkeen Tehtäväryhmä-pudotusvalikkoon

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

marikork committed:
HARJA-905 Muuta alasvetovalikon renderöinti

marikork committed:
HARJA-905 Muuta alasvetovalikon renderöinti

samnes commented on src/cljs/harja/ui/yleiset.cljs:
```diff
@@ -209,54 +209,146 @@ joita kutsutaan kun niiden näppäimiä paineetaan."
+                            (selvita-etaisyydet this sijainti-ylos?))]
+
+    (komp/luo
+      {:component-did-update
```
kun-muuttui - wrapper did-updaten sijaan?

marikork committed:
HARJA-905 Estä näppäimistökäyttö disabloiduilta riveiltä ja muuta kustannussuunnitelmassa suunnitelluissa hankinnoissa valitun toimenpiteen näkyminen

marikork committed:
HARJA-905 Estä näppäimistökäyttö disabloiduilta riveiltä ja muuta kustannussuunnitelmassa suunnitelluissa hankinnoissa valitun toimenpiteen näkyminen

solita-staskila committed:
Korjaa muuttujien nimet, turhat requiret

solita-staskila committed:
Vaihda spanien altit titleksi

solita-staskila committed:
Merge branch 'develop' into HARJA-1180

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

rckopteri committed:
Salli Talvihoitoreittien käyttö jvh:n lisäksi myös ELY_Urakanvalvoja roolilla

rckopteri committed:
Tarkista rooli urakasta

K1ll3rF0x committed:
Korjaile fargate restarttia, jotta se ottaa aimmin asetetun desired
countin huomioon

rckopteri committed:
Muuta käyttöoikeustarkastuksia ely käyttäjälle talvihoitoreitityksissä

samnes committed:
Merge branch 'develop' into HARJA-905_Pudotusvalikkojen_nappaimistokaytto

dependabot[bot] committed:
Bump eclipse-temurin in /aws/fargate

Bumps eclipse-temurin from 17.0.13_11-jre-noble to 17.0.14_7-jre-noble.

---
updated-dependencies:
- dependency-name: eclipse-temurin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

K1ll3rF0x committed:
Merge pull request #3818 from finnishtransportagency/HARJA-1298_korjaile-fargate-restarttia

HARJA-1298 - Korjaile fargate restarttia, jotta se ottaa aimmin asetetun desired

rckopteri committed:
Uusi rooli funktio tarkistamaan onko ely urakanvalvoja urakkaroolissa

rckopteri committed:
Siirrä funktion paikka

rckopteri committed:
Merge pull request #3814 from finnishtransportagency/HARJA-1295-laajenna-talvihoitoreitin-kaytto

HARJA-1295 - Talvihoitoreittit myös ELY_Urakanvalvoja roolille

jarnovayrynen committed:
HARJA-1280 Korjaa päällysteen korjausluokan päättelyä paikkauskohteelle

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1280

samnes commented on src/cljs/harja/ui/yleiset.cljs:
```diff
@@ -209,54 +209,153 @@ joita kutsutaan kun niiden näppäimiä paineetaan."
+                                                               (and maaritelty-id? (.getElementById js/document nappi-id) (not viive-fokuksen-siirtoon?))
+                                                               (.focus (.getElementById js/document nappi-id))
+
+                                                               viive-fokuksen-siirtoon?
```
Testailin tätä kloonia vasten ja en itse huomannut viiveellä vaikutusta fokuksen käyttäytymiseen. Omassa testailussa fokus hyppää disabloituvilla selecteillä bodyyn mutta käyttöä voi käytännössä jatkaa kuitenkin elementin kohdalta joten ei valtavan suuri ongelma. 

En myöskään hoksannut tapauksia missä erikseen määriteltyä id tarvittaisiin kun määrittää  "alasvedon-nappi (.querySelector @valikko-ref "button")" eli ilman when checkkiä.
Jos vielä fokuksen siirtymistä haluaisi parantaa niin joituisi varmaan if:ttelemään component-did-updatessa että muuttuiko disabled arvo. 

Oma ehdotus olisi karsia nämä viiveet ja id välitykset pois jos niitä ei tarvitse ja jättää parannukset toiselle kerralle.

jarnovayrynen committed:
Korjaa testien odotuksia, kun paikk.kohde lisättiin

jarnovayrynen committed:
Korjaa testien odotuksia, kun paikk.kohde lisättiin

jarnovayrynen committed:
Merge pull request #3821 from finnishtransportagency/HARJA-1280

HARJA-1280 Korjaa päällysteen korjausluokan päättelyä paikkauskohteelle

K1ll3rF0x committed:
Merge branch 'develop' into dependabot/docker/aws/fargate/eclipse-temurin-17.0.14_7-jre-noble

K1ll3rF0x committed:
Merge pull request #3820 from finnishtransportagency/dependabot/docker/aws/fargate/eclipse-temurin-17.0.14_7-jre-noble

Bump eclipse-temurin from 17.0.13_11-jre-noble to 17.0.14_7-jre-noble in /aws/fargate

K1ll3rF0x committed:
Merkitse vanhat task definitionit inaktiivisiksi

solita-staskila committed:
Merge branch 'develop' into HARJA-1180

rckopteri committed:
Exceliin ladattaessa kalustojen määrät olivat toteutettu vielä vanhalla tietorakenteella, eikä niiden oikea määrä tullut mukaan exceliin

samnes committed:
Poista disabloitu koodi ja kommentti

samnes committed:
Poista turha require

solita-staskila committed:
Merge pull request #3799 from finnishtransportagency/HARJA-1180

HARJA-1180 Yhdenmukaista raporttinappeja

samnes committed:
Poista käyttämättömät muuttujat

samnes committed:
Poista sivupalkista parametrit joita ei käytössä

samnes committed:
Poista paikkauskohde lomakkeelta käytöstä poistuneet requiret

K1ll3rF0x committed:
Merge branch 'develop' into HARJA-1302_poista-vanhat-task-definitionit

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

K1ll3rF0x committed:
Siirrä shell-koodi run-blokkiin

K1ll3rF0x committed:
Säädä derekisteröinnin toimintaa
* Derekisteröidään vieläkin vanhempi task definition, jotta niitä jää
useampi aktiiviseksi

rckopteri committed:
Tarkenna käyttöoikeuksien tarkistusta

rckopteri committed:
Lisää exceliin parempi teksti, jota käyttäjät eivät koskaa näe.

Exceliä ei voi siis tulostaa, jos reittejä ei ole, mutta parmepi, että korjataan teksti silti.

rckopteri committed:
Merge pull request #3823 from finnishtransportagency/talvihoitoreitti-exel-kalustot-kuntoon

Talvihoitoreitit - lataa kalustomäärä exceliin

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

K1ll3rF0x committed:
Merge branch 'develop' into HARJA-1302_poista-vanhat-task-definitionit

K1ll3rF0x committed:
Merge pull request #3822 from finnishtransportagency/HARJA-1302_poista-vanhat-task-definitionit

Merkitse vanhat task definitionit inaktiivisiksi

samnes committed:
Merge branch 'develop' into HARJA-907-modaalin-saavutettavuus

samnes committed:
Merge pull request #3791 from finnishtransportagency/HARJA-907-modaalin-saavutettavuus

HARJA-907-modaalin-saavutettavuus

K1ll3rF0x committed:
Korjaa /harja/ polun käyttö

K1ll3rF0x committed:
Merge pull request #3824 from finnishtransportagency/korjaa-harjan-domain-polku

HARJA-1309 - Korjaa /harja/ polun käyttö

marikork committed:
HARJA-905 Poista viiveet ja lisätyt id:t fokuksen siirroissa

marikork committed:
Merge branch 'HARJA-905_Pudotusvalikkojen_nappaimistokaytto' of github.com:finnishtransportagency/harja into HARJA-905_Pudotusvalikkojen_nappaimistokaytto
